clang/lib/Sema/SemaTemplateDeduction.cpp

```diff
@@ -2203,6 +2203,76 @@ DeduceTemplateArgumentsByTypeMatch(Sema &S,
 
       return Sema::TDK_NonDeducedMismatch;
     }
+    //     (CHERI clang extension)
+    //
+    //     T __capability
+    case Type::DependentPointer: {
+      const DependentPointerType *PointerParam =
+          cast<DependentPointerType>(Param);
+
+      if (const DependentPointerType *PointerArg =
+              dyn_cast<DependentPointerType>(Arg)) {
+        // Perform deduction on the pointer type.
+        if (Sema::TemplateDeductionResult Result =
+                DeduceTemplateArgumentsByTypeMatch(
+                    S, TemplateParams, PointerParam->getPointerType(),
+                    PointerArg->getPointerType(), Info, Deduced, TDF))
+          return Result;
+
+        // Check that the interpretations are the same
```

Comment:
```suggestion
        // Check that the interpretations are the same.
```

Reply:
Hm, I based this off what was already there:
```
waltham:llvm-project jrtc4% grep '\/\/ Check that the' clang/lib/Sema/SemaTemplateDeduction.cpp
      // Check that the dimensions are the same
        // Check that the interpretations are the same
        // Check that the interpretations are the same
        // Check that the interpretations are the same
  // Check that the deduced argument type is compatible with the original
```

Reply:
Fixed it anyway, those seem to be the exception.